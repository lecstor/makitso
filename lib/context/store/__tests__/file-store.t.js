jest.mock("fs");

const fs = require("fs");
const { promisify } = require("util");

const writeFile = promisify(fs.writeFile);

const Store = require("../file-store");

beforeAll(() => {
  return writeFile("file.json", `{"github":{ "username":"lecstor"}}`, "utf8");
});

describe("File Store", () => {
  const store = Store({ file: "file.json" });

  it("provides get", async () => {
    await expect(store.get("github.username")).resolves.toEqual("lecstor");
  });

  it("provides set", async () => {
    await expect(store.set("twitter.username", "lecstor")).resolves.toEqual(
      "lecstor"
    );
    await expect(store.get("twitter.username")).resolves.toEqual("lecstor");
  });

  it("provides delete", async () => {
    await expect(store.delete("twitter.username")).resolves.toEqual("lecstor");
    await expect(store.get("twitter.username")).resolves.toBeUndefined();
  });
});
