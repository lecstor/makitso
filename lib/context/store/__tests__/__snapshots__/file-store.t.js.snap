// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Store provides delete 1`] = `
Object {
  "github": Object {
    "username": Object {
      "default": "lecstor",
    },
  },
  "twitter": Object {
    "username": Object {},
  },
}
`;

exports[`File Store provides set 1`] = `
Object {
  "github": Object {
    "username": Object {
      "default": "lecstor",
    },
  },
  "twitter": Object {
    "username": Object {
      "default": "lecstor",
    },
  },
}
`;
